src/Assetic/Extension/Twig/TwigResource.php

```diff
@@ -32,7 +32,7 @@ public function __construct(\Twig_LoaderInterface $loader, $name)
     public function getContent()
     {
         try {
-            return $this->loader->getSource($this->name);
+            return $this->loader->getSourceContext($this->name);
```

Comment:
every *core* implementation implements the extra interface. But you can use custom loaders.

What you need is check whether the loader implements the new interface (and then use ``->getSourceContext($this->name)->getCode()`` and fallback to ``getSource()`` otherwise